pkg/skaffold/runner/v1/deploy.go

```diff
@@ -111,6 +111,11 @@ See https://skaffold.dev/docs/pipeline-stages/taggers/#how-tagging-works`)
 		}
 	}
 
+	// if --load-images=true, load all images into cluster
+	if r.runCtx.ForceLoadImages() {
+		localAndBuiltImages = artifacts
```

Comment:
This doesn't seem right: these images may be remote images and they should be pulled into the cluster through the normal mechanism.

I think you've diagnosed that our `localImages` is missing images.  Is the problem with `isImageLocal()` or elsewhere?

Reply:
The issue was that since the images weren't built by this runner, IE they were built by a previous build command and passed as artifacts file, the loop above doesnt append anything to `localAndBuiltImages` (r.wasBuilt returns false). Therefore the images dont get pushed to cluster.

My solution was to let the user tell us that we need to load these images. I see what you are saying about images could be remote, I can move the logic up into the for loop and check `if r.runCtx.ForceLoadImages() || r.wasBuilt` then append to `localAndBuiltImages`.

Does this make more sense since we would only consider local images then?